Sources/Down/AST/Visitors/ListItemPrefixGenerator.swift

```diff
@@ -8,34 +8,43 @@
 
 import Foundation
 
-class ListItemPrefixGenerator {
+/// A ListItemPrefixGenerator is an object used to generate list item prefix.
+///
+/// ListItemPrefixGenerator are created by `ListItemPrefixGeneratorBuilder`, the later used in
+/// conjunction with an instance of `AttributedStringVisitor`.
+public protocol ListItemPrefixGenerator {
+    init(listType: List.ListType, numberOfItems: Int, nestDepth: Int)
+    func next() -> String?
+}
```

Comment:
You could create a convenience initializer as a protocol extension that accepts the `List` node as a single argument, which would clean up the call site a fair bit.